clang/test/SemaSYCL/address-space-parameter-conversions.cpp

```diff
@@ -0,0 +1,59 @@
+// RUN: %clang_cc1 -fsycl-is-device -verify -fsyntax-only -x c++ %s
+
+void bar(int & Data) {}
+void bar2(int & Data) {}
+void bar(__attribute__((opencl_private)) int  & Data) {}
+void foo(int * Data) {}
+void foo2(int * Data) {}
+void foo(__attribute__((opencl_private)) int * Data) {}
+
+template<typename T>
+void tmpl(T *t){}
+
+void usages() {
+  __attribute__((opencl_global)) int *GLOB;
+  __attribute__((opencl_private)) int *PRIV;
+  __attribute__((opencl_local)) int *LOC;
+  int *NoAS;
+
+  bar(*GLOB);
```

Comment:
Since `bar` takes a private pointer (reference), and `*GLOB` is global pointer (reference), we essentially addrspace cast here from global to private. Why this is a positive test case?